### PR TITLE
[REEF-1131] Create, define, and apply VortexAggregationPolicy

### PR DESCRIPTION
**Pull request by afchung**
This addressed the issue by
  * Add VortexAggregatePolicy.
  * Change logic on VortexWorker to aggregate based on VortexAggregatePolicy.
  * Modify example to work with VortexAggregatePolicy

JIRA:
  [REEF-1131](https://issues.apache.org/jira/browse/REEF-1131)

### CODE REVIEW

**Comment by afchung** [comment]
@bgchun @yunseong @johnyangk please have a look when you have time. Thanks!

**Comment by yunseong** [comment]
@afchung Thanks a lot, Andrew! I'll have a look today.

File: lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/evaluator/AggregateContainer.java [hunk 1]
```diff
@@ -94,15 +133,54 @@ public TaskletAggregationRequest getTaskletAggregationRequest() {
...
 
-    return taskletReports.isEmpty() ? Optional.<WorkerReport>empty() : Optional.of(new WorkerReport(taskletReports));
+    // Add to worker report only if there is something to report back.
+    if (!taskletReports.isEmpty()) {
+      workerReportsQueue.addLast(vortexAvroUtils.toBytes(new WorkerReport(taskletReports)));
+      heartBeatTriggerManager.triggerHeartBeat();
+    }
+  }
+
+  public void scheduleTasklet(final int taskletId) {
```
Comment (yunseong) [inline lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/evaluator/AggregateContainer.java]: Simple document will be helpful to understand this method a lot!

File: lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexAggregatePolicy.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+/**
+ * The policy for local aggregation on the {@link org.apache.reef.vortex.evaluator.VortexWorker}s.
+ * The Aggregation function will be triggered on the individual {@link VortexFunction} results on
+ * an "OR" basis of what is specified by the policy.
+ */
+@ClientSide
+@Public
+@Unstable
+public final class VortexAggregatePolicy implements Serializable {
```
Comment (yunseong) [inline lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexAggregatePolicy.java]: Could we add REEF-504, so we can track this class as well?

**Comment by yunseong** [comment]
@afchung Sorry for the late response! I've read the code and this is really great! One question: did you check the `count-based aggregation` works correctly? I think it'd be great to have a test case for the `count-based aggregation`. Thanks!
@bgchun Do you have any other comments?

**Comment by bgchun** [comment]
@afchung @yunseong I think having a count-base aggregation test is great. We can perhaps deal with it in another PR.

My main comment is the use of `Clock` to schedule alarms in the data processing path. `Clock` is the main REEF event loop that processes control events. We probably don't want to use it for creating alarms for data processing. Creating a `ScheduledExecutorService` is good.

**Comment by afchung** [comment]
@bgchun @yunseong Thanks for your reviews! I've addressed your comments, please have another look.

**Comment by bgchun** [comment]
@afchung Looks good. I will test and merge it.

**Pull request closed without merging**